workflows/ampliseq.nf

```diff
@@ -366,7 +380,23 @@ workflow AMPLISEQ {
             if (!params.skip_dada_addspecies) {
                 DADA2_ADDSPECIES ( DADA2_TAXONOMY.out.rds, ch_addspecies, 'ASV_tax_species.tsv' )
                 ch_dada2_tax = DADA2_ADDSPECIES.out.tsv
-            } else { ch_dada2_tax = DADA2_TAXONOMY.out.tsv }
+                if ( params.addsh ) {
+                    VSEARCH_USEARCH_GLOBAL( ch_fasta, ch_assigntax, 'vsearch_blastout.tsv' )
+                    ch_versions = ch_versions.mix(VSEARCH_USEARCH_GLOBAL.out.versions.ifEmpty(null))
+                    ASSIGNSH( ch_dada2_tax, ch_shinfo.collect(), VSEARCH_USEARCH_GLOBAL.out.tsv, 'ASV_tax_species_SH.tsv')
+                    ch_versions = ch_versions.mix(ASSIGNSH.out.versions.ifEmpty(null))
+                    ch_dada2_tax = ASSIGNSH.out.tsv
+                }
+           } else {
+               ch_dada2_tax = DADA2_TAXONOMY.out.tsv
+               if ( params.addsh ) {
+                   VSEARCH_USEARCH_GLOBAL( ch_fasta, ch_assigntax, 'vsearch_blastout.tsv' )
+                   ch_versions = ch_versions.mix(VSEARCH_USEARCH_GLOBAL.out.versions.ifEmpty(null))
+                   ASSIGNSH( ch_dada2_tax, ch_shinfo.collect(), VSEARCH_USEARCH_GLOBAL.out.tsv, 'ASV_tax_SH.tsv')
+                   ch_versions = ch_versions.mix(ASSIGNSH.out.versions.ifEmpty(null))
+                   ch_dada2_tax = ASSIGNSH.out.tsv
+               }
+           }
```

Comment:
I am not sure whether there isnt potential of bugs when a channel is input and output at once, i.e. here `ch_dada2_tax`. Would this work as well (if yes, than I'd prefer this):
```suggestion
           } else {
               if ( params.addsh ) {
                   VSEARCH_USEARCH_GLOBAL( ch_fasta, ch_assigntax, 'vsearch_blastout.tsv' )
                   ch_versions = ch_versions.mix(VSEARCH_USEARCH_GLOBAL.out.versions.ifEmpty(null))
                   ASSIGNSH( DADA2_TAXONOMY.out.tsv, ch_shinfo.collect(), VSEARCH_USEARCH_GLOBAL.out.tsv, 'ASV_tax_SH.tsv')
                   ch_versions = ch_versions.mix(ASSIGNSH.out.versions.ifEmpty(null))
                   ch_dada2_tax = ASSIGNSH.out.tsv
               } else {
                   ch_dada2_tax = DADA2_TAXONOMY.out.tsv
               }
           }
```
also true for the other addition of `ASSIGNSH` below.

Reply:
Thanks, that works and looks better. Changed.